### ISSUE TITLE
Navigation block stuck on loading and importing classic menus

### ISSUE THREAD

**Issue by seothemes** [WordPress/gutenberg#45791]
### Description

This is only happening on the one site and is hard to reproduce.

Each time a menu is selected, WP imports a new nav menu and shows the notice "Classic menu imported successfully". After this, the menu cannot be selected or saved and it gets stuck loading.

When clicking **Manage Menus** and going to the Navigation Menus page, there are thousands of nav menus imported:

<img width="424" alt="Screen Shot 2022-11-16 at 8 59 33 am" src="https://user-images.githubusercontent.com/24793388/202057654-d17a9855-0add-4762-988f-f119591d6c40.png">

This is one of the reasons for #45336 - I don't think nav menus should be forced to use a CPT, it might be better to use simple inner blocks and make the cpt optional.

### Step-by-step reproduction instructions

This only happens on one site - I'm not able to reproduce the issue elsewhere.

1. Add a Navigation block
2. Try to change menus

### Screenshots, screen recording, code snippet

![Navigation block stuck loading](https://user-images.githubusercontent.com/24793388/202056156-85aa0e39-d86b-4dfb-b8a5-bc2d7f211510.gif)

![Trying to change nav menu](https://user-images.githubusercontent.com/24793388/202056669-0ddd4f3b-7630-48c4-8ab6-41ca910ed072.gif)

### Environment info

- WP 6.1.1
- Gutenberg 14.5
- Mac/ubuntu

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by seothemes** [comment]
Deleting all of the nav menus from the db seemed to fix this. Not sure how so many were imported initially. Maybe a check is needed

**Comment by talldan** [comment]
@seothemes Could I ask for some more info?

The screencap shows you clicking an option under 'Import Classic Menu', which will create a new menu. After you've created that menu, what happens? Do you then click to import the menu again? Or is there a bug that continually creates menus without you selecting anything?

Because of the way the gifs loop I have some trouble seeing where they start and end, and I'm not sure what the first gif is showing.

**Comment by seothemes** [comment]
@talldan sorry about the gifs being hard to follow. They are just showing the continuous loading, even without clicking anything in the editor it was happening.

Something caused the Navigation block to keep importing classic menus every time I opened the Site Editor, which ended up being thousands in the db.

After deleting all of the navigation menus the problem was fixed and I haven't been able to reproduce it since.

I have been removing the "id" attribute from the navigation block in pattern HTML to allow them to work correctly on wp.org previews so I'm not sure if that had something to do with it initially.

**Comment by seothemes** [comment]
@talldan this might have been a one off, happy to close the issue unless you think it's worth looking into.

**Comment by mrwweb** [comment]
I don't have many new details to add, but I think I just ran into this issue myself. I ran into this issue while trying to insert a navigation block into a page with the block editor.

- WordPress 6.2.2
- First Custom hybrid theme with `theme.json` that supports Template Parts.
- Subsequently replicated on site with same database with all plugins deactivated and Twenty Twenty Three theme

I tried deleting all menus (of the `wp_navigation` post type) as @seothemes suggested, but that did not resolve the issue.

If relevant, the theme had 5 "classic" menus but only one menu location. The repeatedly imported menu was not the one assigned to the single location, nor the first Alphabetically OR numerically via ID.

Next, I tried to delete the classic menu that was getting repeatedly imported. Not only did that not solve the problem, but the navigation block continued trying to import the old menu and made numerous more copies of it. That makes me wonder if either there's a classic navigation trash that I don't know about or this is somehow a caching issue? I'll update this post if I learn more, but I have yet to figure out a workaround.

As another attempted workaround I tried manually adding the markup for the navigation menu using one of the menus that had been created like this:

`<!-- wp:navigation {"ref": 12345} /-->`

Both this way with a real and madeup `ref` ID still resulted in the constant reloading and creating new menus of the navigation block.

**Comment by mrwweb** [comment]
Useful update: I tried a tremendous number of potential workarounds. Clearing cache, new and old pages, different post types, manual block markup, and more. Nothing helped. It only happens on the staging copy of the site (SiteGround) I have and not a nearly identical local copy.

Finally, I tried installing Gutenberg and v16.1.2 resolved the issue. Unfortunately deactivating Gutenberg brought the issue back (even with the menu inside a reusable block). I haven't been able to test this yet in any 6.3 betas, but I'm hoping it will be resolved there.

**Comment by jordesign** [comment]
Thanks for all this additional testing @mrwweb  - it would be great to have some more testing with the 6.3 Release Candidate, or 6.3 proper once it is released - to see if we can close this.

**Comment by jordesign** [comment]
@mrwweb  With 6.3 released now - I'd love to know if it makes any dfference.

**Comment by mrwweb** [comment]
@jordesign I'm relieved to report that 6.3 does seem to resolve the issue for me. I don't know if that's enough to close the ticket or not since I wasn't the one who opened it.

**Comment by jordesign** [comment]
@seothemes  As this issue appears to be resolved in WordPress 6.3 I'm going to close it. If you're still seeing the issue occur - please feel free to re-open it with additional testing details.
